packages/dd-trace/src/git_properties.js

```diff
@@ -32,4 +39,97 @@ function getGitMetadataFromGitProperties (gitPropertiesString) {
   }
 }
 
-module.exports = { getGitMetadataFromGitProperties, removeUserSensitiveInfo }
+function getRemoteOriginURL (gitConfigContent) {
+  if (!gitConfigContent) {
+    return
+  }
+  const lines = gitConfigContent.split('\n')
+  let index = 0
+
+  // find the remote origin section
+  for (; index < lines.length; index++) {
+    const line = lines[index]
+    if (line[0] !== '[') continue // fast path
+    if (remoteOriginRegex.test(line)) break
+  }
+
+  // find the url key/value in the [remote "origin"] section
+  index++
+  for (; index < lines.length; index++) {
+    const line = lines[index]
+    if (line[0] === '[') return // abort, section didn't contain a url
+    const splitAt = line.indexOf('=')
+    if (splitAt === -1) continue
+    const key = line.slice(0, splitAt).trim().toLowerCase()
+    if (key !== 'url') continue
+    const repositoryUrlValue = line.slice(splitAt + 1).trim()
+    const repositoryUrlMatch = repositoryUrlValue.match(repositoryUrlRegex)
+    if (!repositoryUrlMatch) continue
+    return removeUserSensitiveInfo(repositoryUrlMatch[0])
+  }
+}
+
+function getGitHeadRef (gitHeadContent) {
+  if (!gitHeadContent) {
+    return
+  }
+
+  // Extract the ref after 'ref: '
+  const gitRefMatch = gitHeadContent.match(gitHeadRefRegex)
+
+  if (gitRefMatch?.length > 1) {
+    return gitRefMatch[1]
+  }
+}
+
+function isValidCommitSHA (sha) {
+  if (!sha) {
+    return false
+  }
+  const shaMatch = sha.match(commitSHARegex)
+  if (shaMatch) {
+    return true
+  }
+  return false
+}
+
+function resolveGitHeadSHA (DD_GIT_FOLDER_PATH) {
+  const gitHeadPath = path.join(DD_GIT_FOLDER_PATH, 'HEAD')
+
+  try {
+    const gitHeadContent = fs.readFileSync(gitHeadPath, 'utf8')
+    if (!gitHeadContent) {
+      return
+    }
+
+    const headContent = gitHeadContent.trim()
+
+    // Handle detached head case
+    if (isValidCommitSHA(headContent)) {
```

Comment:
If you call `test` on the regex, it handles everything for you that's done by `isValidCommitSHA` + it doesn't allocate a match object in memory, so it's more performant as well:

```diff
-    if (isValidCommitSHA(headContent)) {
+    if (commitSHARegex.test(headContent)) {
```

If you do the same below for the other calls to `isValidCommitSHA` functions, you can just get rid of the function alltogether :)